--- "zlib-1.2.12_\\crc32.c"	2022-03-27 16:39:42.000000000 -0400
+++ "zlib-1.2.12-patched\\crc32.c"	2022-06-05 15:40:27.077778700 -0400
@@ -1086,7 +1086,7 @@
     uLong crc2;
     z_off_t len2;
 {
-    return crc32_combine64(crc1, crc2, len2);
+    return crc32_combine64(crc1, crc2, (z_off64_t) len2);
 }
 
 /* ========================================================================= */
@@ -1103,11 +1103,11 @@
 uLong ZEXPORT crc32_combine_gen(len2)
     z_off_t len2;
 {
-    return crc32_combine_gen64(len2);
+    return crc32_combine_gen64((z_off64_t) len2);
 }
 
 /* ========================================================================= */
-uLong crc32_combine_op(crc1, crc2, op)
+uLong ZEXPORT crc32_combine_op(crc1, crc2, op)
     uLong crc1;
     uLong crc2;
     uLong op;
