--- "a/win32/Makefile.msc"	2017-10-12 23:27:14.000000000 -0400
+++ "b/win32/Makefile.msc"	2023-10-25 10:28:12.430840100 -0400
@@ -22,13 +22,20 @@
 LD = link
 AR = lib
 RC = rc
-CFLAGS  = -nologo -MD -W3 -O2 -Oy- -Zi -Fd"zlib" $(LOC)
+CFLAGS  = -nologo -W3 -O2 -Oy- -Zi -Fd"zlib" $(LOC)
 WFLAGS  = -D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE
 ASFLAGS = -coff -Zi $(LOC)
 LDFLAGS = -nologo -debug -incremental:no -opt:ref
 ARFLAGS = -nologo
 RCFLAGS = /dWIN32 /r
 
+# use debug version of MSVC CRT for debug builds
+!IFDEF DEBUG
+CFLAGS = $(CFLAGS) /MDd
+!ELSE
+CFLAGS = $(CFLAGS) /MD /DNDEBUG
+!ENDIF
+
 OBJS = adler32.obj compress.obj crc32.obj deflate.obj gzclose.obj gzlib.obj gzread.obj \
        gzwrite.obj infback.obj inflate.obj inftrees.obj inffast.obj trees.obj uncompr.obj zutil.obj
 OBJA =
